libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -95,9 +118,10 @@ public static void TrustServiceUrl(string serviceUrl, DateTime expirationTime)
         /// <returns>True if the host of the service url is trusted; False otherwise.</returns>
         public static bool IsTrustedServiceUrl(string serviceUrl)
         {
-            if (Uri.TryCreate(serviceUrl, UriKind.Absolute, out Uri uri))
+            Uri uri;
+            if (Uri.TryCreate(serviceUrl, UriKind.Absolute, out uri))
             {
-                return IsTrustedUrl(uri);
+                return TrustedUri(uri);
```

Comment:
Strictly speaking, these are URLs. They're locators (that is, they have a concrete location), not URI's. 

I fixed all this some time ago, and it looks like this PR is pulling all the old code forward again.

Reply:
brought your code back

---
In reply to: [198676743](https://github.com/Microsoft/botbuilder-dotnet/pull/702#discussion_r198676743) [](ancestors = 198676743)